### DISCUSSION TITLE
Finding new (prerelease) documentation

### DISCUSSION THREAD

**Discussion by gene96817** [jamulussoftware/jamulus#1866] [Documentation]
There have been some interesting discussion topics regarding documentation. Some of them mention new documentation in progress. Where can I find those work-in-progress. I have some time for testing (and perhaps improving) some docs.

I am beginning a mini project to take a fresh look at simplifying and standardizing a client for a music teacher. I'd be interested in reviewing and testing any ongoing work on this topic.

**Comment by ann0see** [comment]
Move over to the website repo and read the README.md file ;-). If there are any open questions or changes, feel free to raise Pull Requests.

**Reply by gene96817** [reply]
Today is the day to follow through on this. Step 1, install Jekyll... i encountered this error... what is the error message telling me to do?

Fetching jekyll-4.2.0.gem
Successfully installed public_suffix-4.0.6
Successfully installed addressable-2.7.0
Successfully installed colorator-1.1.0
Building native extensions. This could take a while...
ERROR:  Error installing jekyll:
	ERROR: Failed to build gem native extension.

    current directory: /Library/Ruby/Gems/2.6.0/gems/eventmachine-1.2.7/ext
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby -I /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0 -r ./siteconf20210620-1320-oz1wnd.rb extconf.rb
mkmf.rb can't find header files for ruby at /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/include/ruby.h

You might have to install separate package for the ruby development
environment, ruby-dev or ruby-devel for example.

extconf failed, exit code 1

Gem files will remain installed in /Library/Ruby/Gems/2.6.0/gems/eventmachine-1.2.7 for inspection.
Results logged to /Library/Ruby/Gems/2.6.0/extensions/universal-darwin-20/2.6.0/eventmachine-1.2.7/gem_make.out
1 gem installed


Thank you in advance for your reply.

**Reply by ann0see** [reply]
> You might have to install separate package for the ruby development
environment, ruby-dev or ruby-devel for example.

Seems like you‘ll need to install a specific ruby package. I assume it’s possible to get via homebrew: `brew install ruby-devel`?
Also see https://www.google.de/amp/s/stackify.com/install-ruby-on-your-mac-everything-you-need-to-get-going/amp/

or https://gist.github.com/mcls/3118518#rbenv-and-ruby-build

**Reply by gene96817** [reply]
I'll be delayed for a while. Following the trail from more error messages.... ruby won't install without some other commands.... I have to install Fink which makes more commands compatible with macOS (e.g. apt-get). Fink does not yet support Big Sur.

**Reply by ann0see** [reply]
I don’t think there’s a need for fink? You should be able to install everything with brew which is the apt-get equivalent for macOS.

**Reply by gene96817** [reply]
I finally had some time to try brew. I have some error messages and no clue what it wants me to do.

> eugene@MacBook-Air /Applications % brew install ruby-devel
Updating Homebrew...
==> Homebrew is run entirely by unpaid volunteers. Please consider donating:
  https://github.com/Homebrew/brew#donations
==> Auto-updated Homebrew!
Updated 1 tap (homebrew/core).
==> Updated Formulae
Updated 1 formula.

==> Searching for similarly named formulae...
Error: No similarly named formulae found.
Error: No available formula or cask with the name "ruby-devel".
==> Searching for a previously deleted formula (in the last month)...
Error: No previously deleted formula found.
==> Searching taps on GitHub...
Error: No formulae found in taps.
eugene@MacBook-Air /Applications % 

😭.  Help?

**Reply by gene96817** [reply]
Out of boredom and desperation, I ran brew twice more. Each time I get a different result, although it ends with the same seeor messages. What needs to be fixed?

> eugene@MacBook-Air / % brew install ruby-devel                                       
Updating Homebrew...
==> Auto-updated Homebrew!
Updated 2 taps (homebrew/core and homebrew/cask).
==> New Formulae
jello                     lima                      pillow                    pyflow                    terminator
==> Updated Formulae
Updated 136 formulae.
==> New Casks
castr                                       ears                                        hyperkey
==> Updated Casks
Updated 51 casks.

==> Searching for similarly named formulae...
Error: No similarly named formulae found.
Error: No available formula or cask with the name "ruby-devel".
==> Searching for a previously deleted formula (in the last month)...
Error: No previously deleted formula found.
==> Searching taps on GitHub...
Error: No formulae found in taps.
eugene@MacBook-Air / % brew install ruby-devel
==> Searching for similarly named formulae...
Error: No similarly named formulae found.
Error: No available formula or cask with the name "ruby-devel".
==> Searching for a previously deleted formula (in the last month)...
Error: No previously deleted formula found.
==> Searching taps on GitHub...
Error: No formulae found in taps.
eugene@MacBook-Air / %

**Reply by ann0see** [reply]
Probably ruby-devel doesn’t exist for macOS. 

Try to follow https://jekyllrb.com/docs/installation/macos/

**Comment by gilgongo** [comment]
I recently did this PR suggesting re-arranging the docs to introduce a "Server Admin Manual" as a counterpart to the User (formerly "software") Manual https://github.com/jamulussoftware/jamuluswebsite/pull/527 and to make things generally a bit easier to maintain I hope.
